src/plopp/graphics/colormapper.py

```diff
@@ -166,39 +172,37 @@ def rgba(self, data: sc.DataArray) -> np.ndarray:
             colors[one_mask] = self.mask_cmap(self.normalizer(data.values[one_mask]))
         return colors
 
-    def autoscale(self, data: sc.DataArray):
+    def autoscale(self):
         """
-        Re-compute the min and max range of values, given new values.
-        The current range can grow but not shrink.
-
-        Parameters
-        ----------
-        data:
-            The data array containing the values to update the current range.
+        Re-compute the global min and max range of values by iterating over all the
+        artists. If autoscale is set to ``'auto'``, the limits adjust to he current
+        range. If it is set to ``'grow'``, limits can grow but not shrink.
         """
-        vmin, vmax = fix_empty_range(find_limits(data, scale=self.norm))
+        limits = [
+            fix_empty_range(find_limits(artist._data, scale=self.norm))
+            for artist in self.artists.values()
+        ]
+        vmin = reduce(lambda a, b: min(a, b), [v[0] for v in limits])
```

Comment:
Isn't this the same as
```suggestion
        vmin = reduce(min, [v[0] for v in limits])
```
?

Reply:
ah man, let me unnecessarily over-complicate things, your suggestion is too clean for me... ;-)